main/pbx_variables.c

```diff
@@ -662,15 +662,34 @@ void pbx_substitute_variables_helper_full(struct ast_channel *c, struct varshead
 	pbx_substitute_variables_helper_full_location(c, headp, cp1, cp2, count, used, NULL, NULL, 0);
 }
 
+/*! \brief Thread local keeping track of recursion depth */
+AST_THREADSTORAGE(varsub_recurse_level);
+
+#define MAX_VARIABLE_SUB_RECURSE_DEPTH 15
```

Comment:
If the user configures it too high, it will still segfault due to too much recursion, so there would still need to be some hardcoded upper bound, in which case I felt it should just be the maximum level that should be guaranteed to be safe.

In the backtraces I looked at, it segfaulted at under 90 stack frames. Since there are 3 stack frames per recursion, 15 seemed a good number to be safe even if the stack was already pretty deep for some reason.

In reality, I think 15 should be much higher than needed since I can't think why anybody would need to recurse up to 15 times. Most of my use cases recurse only 2, 3, maybe 4 times.

Reply:
Gotcha.  How about just an upgrade note then to alert folks that 15 is the hard limit.